services/wfm/src/main/java/org/openkilda/wfm/topology/event/service/IslService.java

```
@@ -0,0 +1,236 @@
+                .forEach(isls::add);
+
+        for (Isl link: isls) {
+            if (IslStatus.ACTIVE.equals(link.getActualStatus())) {
+                logger.error("Detected ISL {}_{}-{}_{} conflict with {}_{}-{}_{}. Please contact dev team",
+                        link.getSrcSwitchId(), link.getSrcPort(), link.getDestSwitchId(), link.getDestPort(),
+                        isl.getSrcSwitchId(), isl.getSrcPort(), isl.getDestSwitchId(), isl.getDestPort());
+            } else {
+                logger.debug("Skip conflict ISL {}_{}-{}_{} deactivation due to its current status - {}",
+                        link.getSrcSwitchId(), link.getSrcPort(), link.getDestSwitchId(), link.getDestPort(),
+                        link.getActualStatus());
+            }
+        }
+    }
+
+    private Isl getReverseIsl(Isl isl) {
+        Isl reverseIsl = new Isl();
+
+        reverseIsl.setSrcSwitch(isl.getDestSwitch());
+        reverseIsl.setSrcPort(isl.getDestPort());
+        reverseIsl.setDestSwitch(isl.getSrcSwitch());
+        reverseIsl.setDestPort(isl.getSrcPort());
+
+        reverseIsl.setStatus(isl.getStatus());
+
+        return reverseIsl;
+    }
+
+    private Switch createSwitch(SwitchId switchId) {
+        Switch sw = new Switch();
+        sw.setSwitchId(switchId);
+        sw.setStatus(SwitchStatus.INACTIVE);
+        return sw;
+    }
+
+    /**
+     * Isl discovery failed.
+     *
+     * @param isl isl.
+     * @return transaction was successful or not.
+     */
+    public boolean islDiscoveryFailed(Isl isl) {
+        logger.info("ISL {}_{}-{}_{} discovery failed",
+                isl.getSrcSwitchId(), isl.getSrcPort(), isl.getDestSwitchId(), isl.getDestPort());
+        transactionManager.begin();
+        try {
+            processDiscoveryFailedIsl(isl);
+            transactionManager.commit();
+        } catch (Exception e) {
+            transactionManager.rollback();
+            return false;
```

An exception must not be swallowed without a reason.

---

Fixed